### ISSUE TITLE
Error in substitution rule when encountering term + URL + linebreak + URL

### ISSUE THREAD

**Issue by mrjerzy** [jdkato/txtlint#473]
Hi, 

I encountered some unexpected behaviour, which is shown only in a very special situation. 
I defined a substitution rule, which searches for a simple term: 

```
extends: substitution
message: Use '%s' instead of '%s'
level: error
swap:
  'term': Term
```

Then I defined some text, which must be exactly as follow: 
- It must contain the term
- then a URL containing the same term (can be in the same line as the first term)
- at least one line break
- the exact same URL again 

The substitution is unexpectedly reported in the URL, not in the first word "term". 
```
term                            <- i'd expect the substitution to be applied here, ....
https://abc.com/term

https://abc.com/term            <- ..., but the substitution is applied here. the word "term" in the URL is reported to be bad
```

The CLI reports following:
```
 some-path/error.md
 4:17  error  Use 'Term' instead of 'term'  Rules.test 
```

The rule works as expected, when you add following change to above example: 
- remove the line break between the two URLs 
- make the URLs different (for example remove an 'a' from any of the URL)
- remove one of both URLs
- remove the term from at least one of the both URLs

Any of above changes lead to following outout (which is correct in my opinion)
```
 some-path/error.md
 1:1  error  Use 'Term' instead of 'term'  Rules.test 
```

The issue feels kind of similar to https://github.com/errata-ai/vale/issues/455, but I am not able to tell that for sure.

## Setup
- macOS Monterey, 12.3.1 
- I installed vale through the nix package manager
- vale version 2.20.0

## .vale.ini

```
StylesPath = styles

[*.md]
BasedOnStyles = Vale, Rules
```

## Folder Structure 

```
styles/Rules/test.yml (contains substitution shown above)
some-path/error.md
.vale.ini
```

## vale Command

I execute `vale ./some-path/*`
